### PR TITLE
[AiLab] Delete models button

### PR DESCRIPTION
**Pull request by Erin007**
Follow up to #39844

Adds a delete button the Model Manager Dialog that, when clicked, opens a confirmation dialog to confirm deletion of the selected model. 

https://user-images.githubusercontent.com/12300669/113876927-35441c00-9786-11eb-974d-4d3137ab9ed9.mov

If the App Lab level has a levelbuilder trained model, we hide the delete button to prevent users from deleting that model. 
Screen Shot 2021-04-06 at 3 07 31 PM

### CODE REVIEW

File: dashboard/app/controllers/api/v1/ml_models_controller.rb [hunk 1]
```diff
@@ -27,16 +27,16 @@ def save
   end
 
   # GET api/v1/ml_models/names
-  # Retrieve the names and ids of a user's trained ML models.
+  # Retrieve the names, ids and metadata of a user's trained ML models.
   def names
-    user_ml_model_data = UserMlModel.where(user_id: current_user&.id).map {|user_ml_model| {"id": user_ml_model.model_id, "name": user_ml_model.name}}
+    user_ml_model_data = UserMlModel.where(user_id: current_user&.id).map {|user_ml_model| {"id": user_ml_model.model_id, "name": user_ml_model.name, "metadata": JSON.parse(user_ml_model.metadata)}}
```
Comment (breville) [inline dashboard/app/controllers/api/v1/ml_models_controller.rb]: nit: are the quotes needed?

File: dashboard/config/scripts/levels/AppLab AI Enabled.level [hunk 2]
```diff
@@ -52,17 +52,19 @@
     "palette_category_at_start": "data",
     "validation_enabled": "false",
     "widget_mode": "false",
-    "libraries_enabled": "false",
+    "libraries_enabled": "true",
```
Comment (breville) [inline dashboard/config/scripts/levels/AppLab AI Enabled.level]: Why are libraries enabled?

File: apps/src/code-studio/components/ModelManagerDialog.jsx [hunk 3]
```diff
@@ -104,9 +164,52 @@ export default class ModelManagerDialog extends React.Component {
...
+          <div style={styles.left}>
+            <p style={styles.message}>
+              This model will be permanently deleted, and you will not be able
+              to use this model in any App Lab projects.
+            </p>
+            <div>
+              <Button
+                text={'No'}
+                color={Button.ButtonColor.orange}
+                onClick={this.closeConfirmDialog}
```
Comment (breville) [inline apps/src/code-studio/components/ModelManagerDialog.jsx:215]: Pressuming "No" closes the entire dialog, but I think it should actually go back to the previous dialog state in which you can choose to import/delete.

File: apps/src/code-studio/components/ModelManagerDialog.jsx [hunk 4]
```diff
@@ -67,21 +101,47 @@ export default class ModelManagerDialog extends React.Component {
...
+  };
+
+  deleteModel = () => {
+    $.ajax({
+      url: `/api/v1/ml_models/${this.state.selectedModel.id}`,
+      method: 'DELETE'
+    }).then(response => {
+      this.setState({
+        deletionStatus: response.status,
+        confirmDialogOpen: false
```
Comment (made-line) [inline apps/src/code-studio/components/ModelManagerDialog.jsx]: Similar to Brendan's comment about the 'No' button, I wonder if the user should be redirected to the beginning of the dialog prompt after deleting a model. This way, they can visually confirm it's deleted and they can perform another action without having to go back to the settings.

**Comment by Erin007** [comment]
Updated to show the initial delete/import dialog when no is clicked in the confirmation dialog. 

https://user-images.githubusercontent.com/12300669/114088705-4a52a500-9883-11eb-9793-a9153813e429.mov

**Comment by Erin007** [comment]
Screen Shot 2021-04-08 at 3 47 59 PM
Also included a warning if the user's model fails to delete.

File: apps/src/code-studio/components/ModelManagerDialog.jsx [hunk 5]
```diff
@@ -67,21 +110,49 @@ export default class ModelManagerDialog extends React.Component {
...
+        this.setState({confirmDialogOpen: false});
+      }
+    });
   };
 
   render() {
     const {isOpen} = this.props;
-    const noModels = this.state.models.length === 0;
+    const noModels =
+      !this.state.isModelListPending && this.state.models.length === 0;
```
Comment (Erin007) [inline apps/src/code-studio/components/ModelManagerDialog.jsx:143]: @breville this fixes the weirdness we saw where the warning popped up before the models were loaded

File: apps/src/code-studio/components/ModelManagerDialog.jsx [hunk 6]
```diff
@@ -94,7 +165,11 @@ export default class ModelManagerDialog extends React.Component {
                 </option>
               ))}
             </select>
-            {noModels && <div>You have not trained any A.I. models yet.</div>}
+            {noModels && (
+              <div style={styles.message}>
```
Comment (Erin007) [inline apps/src/code-studio/components/ModelManagerDialog.jsx:169]: And adding styles here fixed the behavior where the message would disappear and re-appear with mouse movement.

**Review by made-line — APPROVED** [review]
LGTM!

**Comment by breville** [comment]
Sorry for the delayed followup.  It feels as though deletion should have a spinner for slow network connections, and that after deleting, we should return to the updated model list again.

**Pull request merged**